### PR TITLE
chore: Update iOS dependencies

### LINKED ISSUE #669: a newer version of google_mlkit_text_recognition that supports GoogleDataTransport (~> 10.0)
a newer version of google_mlkit_text_recognition that supports GoogleDataTransport (~> 10.0).

Firebase just update their package requires GoogleDataTransport (~> 10.0) , however google_mlkit_text_recognition depends on an earlier version of GoogleDataTransport (< 10.0). it caused conflict . I'm unable to install my pod commands in project.

### PR DESCRIPTION
**Pull request by OutdatedGuy**
fixes #669

## Changes
- Updates `GoogleMLKit` to [v7.0.0](https://developers.google.com/ml-kit/release-notes#september_30_2024) released on 30th September.
- Sets Minimum iOS Deployment Target to `15.5.0`

### CODE REVIEW

**Comment by fbernaly** [comment]
@OutdatedGuy: thanks for your contributions, I am going to review your PR. Did you test the changes with the example app?

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,110 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 2024-10-04
```
Comment (fbernaly) [inline CHANGELOG.md:6]: Change date to 2024-10-07

**Review by fbernaly — CHANGES_REQUESTED** [review]

**Comment by fbernaly** [comment]
Do we need to bump all the dependencies from iOS 12 to 15.5.0?

File: AUTHORS [hunk 2]
```diff
@@ -5,3 +5,4 @@
 
 Francisco Bernal <fbernaly@gmail.com>
 Bharat Biradar <biradar.1.bharat@gmail.com>
+Om Phatak <everythingoutdated@gmail.com>
```
Comment (fbernaly) [inline AUTHORS:8]: Thanks for your contributions. Once this PR is merged, you will be listed as a contributor as many others have done. This file is only for those who have written a plug-in from scratch: a whole new ML kit feature/API from scratch. Please, revert the changes in this file. Thanks.

**Review by fbernaly — CHANGES_REQUESTED** [review]

**Comment by OutdatedGuy** [comment]
> @OutdatedGuy: thanks for your contributions, I am going to review your PR. Did you test the changes with the example app?

Yeah, I have tested it on iOS simulator and a real device. Everything seems to be working fine.

**Comment by OutdatedGuy** [comment]
> Do we need to bump all the dependencies from iOS 12 to 15.5.0?

Most likely, as the [official release notes](https://developers.google.com/ml-kit/release-notes#september_30_2024) mentioned the minimum support change.

I tried running few plugins on iOS 13, but got error due to minimum support incompatible.

**Comment by vietstone-ng** [comment]
Expect this to be released. Thank you all very much.

**Comment by vietstone-ng** [comment]
@vallerydelexy  The bump is acceptable.
- First, it's from native, as @OutdatedGuy pointed out above (https://developers.google.com/ml-kit/release-notes#september_30_2024)
- Another reason may be iOS's privacy manifest, as old versions do not have it.

**Comment by poka-IT** [comment]
I don't see any change log about septembre 2024 here: https://developers.google.com/ml-kit/release-notes#september_30_2024

Am I cross-eyed?

The bump is really hard...

**Review by fbernaly — APPROVED** [review]

**Comment by fbernaly** [comment]
Screenshot 2024-10-07 at 11 03 05 AM

**Pull request merged**